integration_test/go_java_launcher_integration_test.go

```diff
@@ -114,8 +143,21 @@ func TestSubProcessesParsedMonitorSignals(t *testing.T) {
 	assert.Len(t, trapped.FindAll(output.Bytes(), -1), 2, "expect two messages that SIGPOLL was caught")
 }
 
-func runMainWithArgs(t *testing.T, staticConfigFile, customConfigFile string) (string, error) {
-	output, err := mainWithArgs(t, staticConfigFile, customConfigFile).CombinedOutput()
+func runMainWithArgs(t *testing.T, staticConfigFile, customConfigFile string, env ...string) (string, error) {
+	jdkDir := "jdk"
+	javaHome, err := filepath.Abs(jdkDir)
+	if err != nil {
+		log.Fatalf("Failed to calculate absolute path of '%s': %v\n", jdkDir, err)
+	}
+
+	// Override existing environment when running subprocess.
+	var customEnv []string
+	customEnv = append(customEnv, "JAVA_HOME="+javaHome)
+	customEnv = append(customEnv, env...)
```

Comment:
note: you can do this in 1 line: `customEnv := append([]string{"JAVA_HOME="+javaHome}, env...)`

Reply:
Done